modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp

```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
-			<aui:option label="none" selected="<%= true %>" value="" />
-
-			<%
-			for (LayoutSetPrototype curLayoutSetPrototype : layoutSetPrototypes) {
-				UnicodeProperties settingsProperties = curLayoutSetPrototype.getSettingsProperties();
-			%>
-
-				<aui:option data-servletContextName='<%= settingsProperties.getProperty("customJspServletContextName", StringPool.BLANK) %>' value="<%= curLayoutSetPrototype.getLayoutSetPrototypeId() %>"><%= HtmlUtil.escape(curLayoutSetPrototype.getName(locale)) %></aui:option>
-
-			<%
-			}
-			%>
-
-		</aui:select>
-
-		<c:if test="<%= !siteGoup.isStaged() %>">
+<h4 class="sheet-subtitle"><liferay-ui:message key="public-pages" /></h4>
+
+<aui:field-wrapper cssClass="form-group">
+	<c:choose>
+		<c:when test="<%= (publicLayoutSetPrototype == null) && (siteGoup.getPublicLayoutsPageCount() == 0) && !layoutSetPrototypes.isEmpty() %>">
+			<c:if test="<%= disableLayoutSetPrototypeInput %>">
+				<div class="alert alert-info">
+					<liferay-ui:message key="you-cannot-apply-a-site-template-because-you-modified-the-display-settings-of-this-site" />
+				</div>
+			</c:if>
+
+			<aui:select disabled="<%= disableLayoutSetPrototypeInput %>" helpMessage="site-templates-with-an-incompatible-application-adapter-are-disabled" label="site-template" name="publicLayoutSetPrototypeId">
+				<aui:option label="none" selected="<%= true %>" value="" />
+
+				<%
+				for (LayoutSetPrototype curLayoutSetPrototype : layoutSetPrototypes) {
+					UnicodeProperties settingsProperties = curLayoutSetPrototype.getSettingsProperties();
+				%>
+
+					<aui:option data-servletContextName='<%= settingsProperties.getProperty("customJspServletContextName", StringPool.BLANK) %>' value="<%= curLayoutSetPrototype.getLayoutSetPrototypeId() %>"><%= HtmlUtil.escape(curLayoutSetPrototype.getName(locale)) %></aui:option>
+
+				<%
+				}
+				%>
+
+			</aui:select>
+
+			<c:if test="<%= !siteGoup.isStaged() %>">
+				<c:choose>
+					<c:when test="<%= hasUnlinkLayoutSetPrototypePermission %>">
+						<div class="hide" id="<portlet:namespace />publicLayoutSetPrototypeIdOptions">
+							<c:if test="<%= disableLayoutSetPrototypeInput %>">
+								<div class="alert alert-info">
+									<liferay-ui:message key="you-cannot-enable-the-propagation-of-changes-because-you-modified-the-display-settings-of-this-site" />
+								</div>
+							</c:if>
+
+							<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" helpMessage="enable-propagation-of-changes-from-the-site-template-help" inlineLabel="right" label="enable-propagation-of-changes-from-the-site-template" labelCssClass="simple-toggle-switch" name="publicLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= publicLayoutSetPrototypeLinkEnabled %>" />
+						</div>
+					</c:when>
+					<c:otherwise>
+						<aui:input name="publicLayoutSetPrototypeLinkEnabled" type="hidden" value="<%= true %>" />
```

Comment:
This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation: